src/game/logic/Board.java

```diff
@@ -65,6 +65,25 @@ public Cell getCell(int index){
     public void setCell(int index, PlayerToken player){
         cells[index].setPlayer(player);
     }
+    public int getCount(PlayerToken player) {
+        int cnt = 0;
+        for(Cell cell: cells) {
+            if(cell.isOccupiedBy(player))
+                cnt++;
+        }
+        return cnt;
+    }
+    public boolean moveFromTo(int srcIndex, int destIndex, PlayerToken player){
+        if (getCell(destIndex).isOccupied())
+            return false;
+        try {
+            setCell(destIndex, player ); //set dest = source cell, get index, make sure both indices exist
+            setCell(srcIndex, PlayerToken.NOPLAYER); }
+        catch(Exception e)  {
```

Comment:
This is bad practice and we should not be catching all exceptions. We should only accept what we expect to encounter or know how to deal with. The error we want to catch here is `ArrayIndexOutOfBoundsException`